client/src/partials/enterprises/enterprises.js

```diff
@@ -31,65 +30,161 @@ function EnterpriseController(Enterprises, Currencies, util, Notify) {
       return;
     }
 
     var promise;
-    var creation = (vm.view === 'create');
-    var enterprise = angular.copy(vm.enterprise);
+    var creation = (vm.hasEnterprise === false);
+    var enterprise = cleanEnterpriseForm(vm.enterprise);
 
     promise = (creation) ?
       Enterprises.create(enterprise) :
       Enterprises.update(enterprise.id, enterprise);
 
     return promise
       .then(function () {
-        return refreshEnterprises();
-      })
-      .then(function () {
-        update(enterprise.id);
-        Notify.success(creation ? 'FORM.INFO.SAVE_SUCCESS' : 'FORM.INFO.UPDATE_SUCCESS');
+        Notify.success(creation ? 'FORM.INFOS.SAVE_SUCCESS' : 'FORM.INFOS.UPDATE_SUCCESS');
       })
       .catch(Notify.handleError);
   }
 
+  function cleanEnterpriseForm(enterprise) {
+    return {
+      id : enterprise.id,
+      name : enterprise.name,
+      abbr : enterprise.abbr,
+      phone : enterprise.phone,
+      email : enterprise.email,
+      location_id : enterprise.location_id,
+      currency_id : enterprise.currency_id,
+      po_box : enterprise.po_box,
+      gain_account_id : enterprise.gain_account.id,
+      loss_account_id : enterprise.loss_account.id
+    };
+  }
+
+  /* ================================ PROJECT ================================ */
+
+  vm.addProject = addProject;
+  vm.editProject = editProject;
+  vm.deleteProject = deleteProject;
+
+  // refresh the displayed projects
+  function refreshProjects() {
+    return Projects.read(null, { complete : 1 })
+      .then(function (projects) {
+        vm.projects = projects;
+      });
+  }
+
+  /**
+   * @function editProject
+   * @desc launch project modal for editing
+   */
+  function editProject(id) {
+    var params = {
+      action : 'edit',
+      identifier : id,
+      enterprise : vm.enterprise
+    };
+    Modal.openProjectActions(params)
+    .then(function (value) {
+      if (!value) { return; }
+
+      refreshProjects();
+      Notify.success('FORM.INFOS.UPDATE_SUCCESS');
+    })
+    .catch(Notify.handleError);
+  }
+
+  /**
+   * @function addProject
+   * @desc launch project modal for adding new
+   */
+  function addProject() {
+    var params = {
+      action : 'create',
+      enterprise : vm.enterprise
+    };
+    Modal.openProjectActions(params)
+    .then(function (value) {
+      if (!value) { return; }
+
+      refreshProjects();
+      Notify.success('FORM.INFOS.CREATE_SUCCESS');
+    })
+    .catch(Notify.handleError);
+  }
+
+  /**
+   * @function deleteProject
+   * @desc delete an existing project
+   * @param {number} id The project id
+   */
+  function deleteProject(id, pattern) {
+    var params = {
+      pattern: pattern,
+      elementName: $translate.instant('FORM.LABELS.PROJECT')
+    };
+
+    Modal.openConfirmDeletion(params)
+    .then(function (bool) {
+      if (!bool) { return; }
+
+      Projects.delete(id)
+        .then(function () {
+          Notify.success('FORM.INFOS.DELETE_SUCCESS');
```

Comment:
Since @sfount merged https://github.com/IMA-WorldHealth/bhima-2.X/pull/616, this should be `FORM.INFO.DELETE_SUCCESS`.